src/main/scala/scala/collection/decorators/package.scala

```diff
@@ -8,13 +8,13 @@ package object decorators {
   implicit def iteratorDecorator[A](it: Iterator[A]): IteratorDecorator[A] =
     new IteratorDecorator[A](it)
 
-  implicit def IterableDecorator[C](coll: C)(implicit it: IsIterable[C]): IterableDecorator[C, it.type] =
+  implicit def iterableDecorator[C](coll: C)(implicit it: IsIterable[C]): IterableDecorator[C, it.type] =
     new IterableDecorator(coll)(it)
 
-  implicit def SeqDecorator[C](coll: C)(implicit seq: IsSeq[C]): SeqDecorator[C, seq.type] =
+  implicit def seqDecorator[C](coll: C)(implicit seq: IsSeq[C]): SeqDecorator[C, seq.type] =
     new SeqDecorator(coll)(seq)
 
-  implicit def MapDecorator[C](coll: C)(implicit map: IsMap[C]): MapDecorator[C, map.type] =
+  implicit def mapDecorator[C](coll: C)(implicit map: IsMap[C]): MapDecorator[C, map.type] =
```

Comment:
Changing these names is required for it to compile with Scala 3 (the name of implicit def can't clash with the existing class). I understand this breaks the compatibility so I'm leaving the decision here to the maintainers.

Options:
1. Change the names, break the compatibility, don't bump the major version
2. Change the names, break the compatibility, bump the major version
3. Change the names only for scala 3 (create a dedicated file for scala 3)

Reply:
Hello @povder, thank you for your contribution!

Option 1 is not practicable. We have to bump the major version number if we break the binary compatibility.

I am not sure about option 3: I believe it should be safe because the Scala 3 artifacts have no previous release, and users can not depend on both Scala 2 and Scala 3 artifacts in the same build. I wonder if this can create issues somehow, still… Also, I am a bit worried by the introduced complexity.

Last thought: it would be a good opportunity to move to 1.x.y versions instead of the current 0.x.y.

Thougths?

Reply:
Thanks for your input @julienrf. My preferred solution is to change the names for both Scala 2 and 3, keep the single file and bump version to 1.0.0.

Reply:
I don't take it as given that this repo should _ever_ go to 1.0.0. It seems plausible to me that it could stay in 0.x versions forever, with no bincompat guarantees at all.

It depends on whether you see the purpose of this library as making user-contributed code available for use in production, or just to make it available for people to try out. And I'm not sure we've ever really decided that.

Reply:
OK, so maybe another solution would be to go with a version 0.3.0, but I wonder what would be the problem of using 1.0.0 instead of 0.3.0? The only issue I see with 0.3.0 is that it won’t be possible to distinguish between a future release that breaks source compatibility and a future release that breaks binary compatibility (they would both have the number 0.4.0).

Reply:
I'm thinking more of the social aspect than the technical aspect. Version numbers send signals to human beings as well as to tooling, and staying on 0.x indefinitely helps communicate that this repo is just a pile of experimental code that may or may not have been seriously tested or vetted by anybody.

Reply:
@SethTisue I think people may still expect releases of this lib to be binary compatible on minor version bumps even if it hasn't ever been publicly communicated because `versionPolicyIntention := Compatibility.BinaryCompatible` is set.